trollmoves/tests/test_dispatcher.py

```diff
@@ -397,17 +399,124 @@ def test_create_dest_url():
             msg.data = {'sensor': 'viirs', 'product': 'green_snow', 'platform_name': 'NOAA-20',
                         'start_time': datetime(2019, 9, 19, 9, 19), 'format': 'tif'}
             # SSH protocol, no username
-            url, params = dp.create_dest_url(msg, 'target2', config['target2'])
+            url, params, client = dp.create_dest_url(msg, 'target2',
+                                                     config['target2'])
             expected_url = "ssh://server.target2.com/satellite/viirs/sat_201909190919_NOAA-20.tif"
             assert url == expected_url
             assert params == {'ssh_key_filename': '~/.ssh/rsa_id.pub'}
+            assert client == "target2"
 
             # SCP protocolw with username
-            url, params = dp.create_dest_url(msg, 'target3', config['target3'])
+            url, params, client = dp.create_dest_url(msg, 'target3',
+                                                     config['target3'])
             expected_url = "scp://user@server.target2.com/satellite/viirs/sat_201909190919_NOAA-20.tif"
             assert url == expected_url
+            assert client == "target3"
 
     finally:
         if dp is not None:
             dp.close()
         os.remove(config_file_name)
+
+
+test_yaml_pub = test_yaml2 + """
+target3:
+  host: scp://user@server.target2.com
+  filepattern: 'sat_{start_time:%Y%m%d%H%M}_{platform_name}.{format}'
+  directory: /satellite/{sensor}
+  publish_topic: "/topic/{platform_name}"
+  dispatch_configs:
+    - topics:
+        - /level2/viirs
+"""
+@patch('trollmoves.dispatcher.Message')
+@patch('trollmoves.dispatcher.ListenerContainer')
+@patch('trollmoves.dispatcher.NoisyPublisher')
+def test_publisher(NoisyPublisher, ListenerContainer, Message):
+    """Test the publisher is initialized."""
+    pub = Mock()
+    NoisyPublisher.return_value = pub
+    try:
+        with NamedTemporaryFile('w', delete=False) as config_file:
```

Comment:
Why not letting `delete=True` ?

Reply:
Because the file would be deleted when _closed_, not when the context manager exits. So when the file is written to and closed, the context manager will delete it and the config can't be read.

Reply:
Ah, good point! that's an important detail from the documentation :)